packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/index.jsx

```diff
@@ -46,6 +48,26 @@ export const StorefrontPreview = ({
         experimentalUnsafeAdditionalSearchParams,
         experimentalUnsafeReloadServerSide
     ])
+
+    useEffect(() => {
+        if (enabled && isHostTrusted) {
+            const originalWillSendRequest = apiClients.willSendRequest
```

Comment:
FYI the existing mechanism has this `willSendRequest`, which will get called before requesting SCAPI: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/eab57e523f2a8b2bf961d24a51a7fbe9f8310591/packages/template-retail-react-app/app/commerce-api/index.js#L147-L151

By default, `willSendRequest` will automatically inject the access token into the request header.